core/thread/src/TThreadExecutor.cxx

```diff
@@ -1,15 +1,32 @@
 #include "ROOT/TThreadExecutor.hxx"
 #include "tbb/tbb.h"
+#include "TROOT.h"
+#include <iostream>
 
 namespace ROOT{
+
+  unsigned TThreadExecutor::fPoolSize = 0;
+
   TThreadExecutor::TThreadExecutor():fInitTBB(new tbb::task_scheduler_init()){
   }
 
   TThreadExecutor::TThreadExecutor(size_t nThreads):fInitTBB(new tbb::task_scheduler_init(nThreads)){
+    if (fPoolSize != 0){
+      auto cause = "TThreadExecutor";
+      if(ROOT::IsImplicitMTEnabled()){
+        fPoolSize = ROOT::GetImplicitMTPoolSize();
+        fPoolSize = 3;
```

Comment:
Why do you call `fPoolSize = ROOT::GetImplicitMTPoolSize();` before setting it to 3? If the former has a side effect then adding a comment would be nice.